shopfloor_reception/services/reception.py

```diff
@@ -599,47 +599,50 @@ def _set_quantity__by_packaging(self, picking, selected_line, packaging):
+
+        """
         pack_location = package.location_id
-        if pack_location:
-            (
-                move_dest_location_ok,
-                pick_type_dest_location_ok,
-            ) = self._check_location_ok(pack_location, selected_line, picking)
-            if not (move_dest_location_ok or pick_type_dest_location_ok):
-                # If the scanned package has a location that isn't a child
-                # of the move dest, return an error
-                message = self.msg_store.dest_location_not_allowed()
-                return self._response_for_set_quantity(
-                    picking, selected_line, message=message
-                )
-            quantity = selected_line.qty_done
-            __, qty_check = selected_line._split_qty_to_be_done(
-                quantity,
-                lot_id=False,
-                shopfloor_user_id=False,
-                expiration_date=False,
+        if not pack_location:
+            line.result_package_id = package
+            return None
+        (
+            move_dest_location_ok,
+            pick_type_dest_location_ok,
+        ) = self._check_location_ok(pack_location, line, picking)
+        if not (move_dest_location_ok or pick_type_dest_location_ok):
+            # Package location is not a child of the move destination
+            message = self.msg_store.dest_location_not_allowed()
+            return self._response_for_set_quantity(picking, line, message=message)
+        quantity = line.qty_done
+        __, qty_check = line._split_qty_to_be_done(
+            quantity,
+            lot_id=False,
+            shopfloor_user_id=False,
+            expiration_date=False,
+        )
+        if qty_check == "greater":
+            return self._response_for_set_quantity(
+                picking,
+                line,
+                message=self.msg_store.unable_to_pick_more(line.product_uom_qty),
             )
-            if qty_check == "greater":
-                return self._response_for_set_quantity(
-                    picking,
-                    selected_line,
-                    message=self.msg_store.unable_to_pick_more(
-                        selected_line.product_uom_qty
-                    ),
-                )
-            # If the scanned package has a valid destination,
-            # set both package and destination on the package.
-            selected_line.result_package_id = package
-            selected_line.location_dest_id = pack_location
+        line.result_package_id = package
+        line.location_dest_id = pack_location
 
-            response = self._post_line(selected_line)
-            if response:
-                return response
+    def _set_quantity__by_package(self, picking, selected_line, package):
+        response = self._set_package_on_move_line(picking, selected_line, package)
+        if response:
+            return response
+        if package.location_id:
```

Comment:
I still think you lost the call to `_post_line` when scanning a package that is already on a location
```suggestion
        if package.location_id:
            response = self._post_line(selected_line)
            if response:
                return response
```